iris-mpc/src/server/mod.rs

```diff
@@ -811,3 +810,92 @@ pub async fn server_main(config: Config) -> eyre::Result<()> {
     }
     Ok(())
 }
+
+async fn prepare_stores(config: &Config) -> Result<(Store, GraphPg<Aby3Store>), Report> {
+    let schema_name = format!(
+        "{}_{}_{}",
+        config.app_name, config.environment, config.party_id
+    );
+
+    // Always need the "GPU" database config
+    let gpu_db_config = config
+        .database
+        .as_ref()
+        .ok_or(eyre!("Missing GPU database config"))?;
+
+    let gpu_postgres_client = PostgresClient::new(&gpu_db_config.url, &schema_name).await?;
+
+    match config.mode_of_deployment {
+        ModeOfDeployment::ShadowIsolation => {
+            // This mode uses only CPU DB
+            let cpu_db_config = config
+                .cpu_database
+                .as_ref()
+                .ok_or(eyre!("Missing CPU database config in ShadowIsolation"))?;
+            let cpu_postgres_client = PostgresClient::new(&cpu_db_config.url, &schema_name).await?;
+
+            // Store -> CPU
+            tracing::info!(
+                "Creating new iris store from: {:?} in mode {:?}",
+                cpu_db_config,
+                config.mode_of_deployment
+            );
+            let store = Store::new(&cpu_postgres_client).await?;
+
+            // Graph -> CPU
+            tracing::info!(
+                "Creating new graph store from: {:?} in mode {:?}",
+                cpu_db_config,
+                config.mode_of_deployment
+            );
+            let graph_store = GraphStore::new(&cpu_postgres_client).await?;
+
+            Ok((store, graph_store))
+        }
+
+        ModeOfDeployment::ShadowReadOnly => {
+            // Store -> GPU
+            tracing::info!(
+                "Creating new iris store from: {:?} in mode {:?}",
+                gpu_db_config,
+                config.mode_of_deployment
+            );
+            let store = Store::new(&gpu_postgres_client).await?;
+
+            // Graph -> CPU
+            let cpu_db_config = config
+                .cpu_database
+                .as_ref()
+                .ok_or(eyre!("Missing CPU database config in ShadowReadOnly"))?;
+            let cpu_postgres_client = PostgresClient::new(&cpu_db_config.url, &schema_name).await?;
+
+            tracing::info!(
+                "Creating new graph store from: {:?} in mode {:?}",
+                cpu_db_config,
+                config.mode_of_deployment
+            );
+            let graph_store = GraphStore::new(&cpu_postgres_client).await?;
+
+            Ok((store, graph_store))
+        }
+
+        // All other modes: we only need/use the GPU DB
```

Comment:
The name GPU DB is not really representative here, right? Even in the standard deployment we do not have a "GPU DB" here, this would still be the DB for the CPU deployment, no? The GPU binary never calls this method

I think ATM out of the 3 cases only 2 are ever happening:
* SharesDB and GraphDB are the same DB => Isolation mode (DB_URL points to its own DB), Standard Mode (DB_URL points to production DB)
* SharesDB and GraphDB are different DBs => Shadow Mode (DB_URL points to production DB with readonly access, CPU_DB_URL points to its own DB)